File: 8kyu/8kyu 2/squareSum.js
function squareSum(numbers) {
   const result = [];
   numbers.forEach(element => {
      result.push(Math.pow(element, 2));
   });
   return result.length > 0 ? result.reduce((acc, curr) => acc + curr) : 0;
}


function squareSum(numbers) {
   return numbers.reduce((sum, num) => sum + (num * num), 0);
}


function squareSum(numbers) {
   var sum = 0;
   numbers.forEach(function (n) {
      sum += n * n
   });
   return sum;
}


console.log(squareSum([1, 2]), 5)
console.log(squareSum([0, 3, 4, 5]), 50)

File: Python/8py 3/square_sum.py
"""
|--------------------------------------------------------------------------
| Square(n) Sum
|--------------------------------------------------------------------------
|
"""

def square_sum(numbers):  
   return sum(map(lambda i: i**2, numbers))


def square_sum(numbers):
   res = 0
   for num in numbers:
      res = res + num*num
   return res


def square_sum(numbers):
   return sum(x ** 2 for x in numbers)


def square_sum(numbers):
   result = []
   for sqr in numbers:
      result.append(sqr ** 2)
   return sum(result)


print(square_sum([1,2]), 5)
print(square_sum([0, 3, 4, 5]), 50)
print(square_sum([]), 0)
print(square_sum([-1,-2]), 5)
print(square_sum([-1,0,1]), 2)